src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -277,13 +277,19 @@ protected virtual bool IsAttributeConditionallyOmitted(NamedTypeSymbol attribute
                 ArrayBuilder<BoundExpression> boundNamedArgumentsBuilder = null;
                 HashSet<string> boundNamedArgumentsSet = null;
 
+                // VariableUsePass assumes we're not in a scope that can reference local variables.
+                // Need to rework VariableUsePass if that ever changes.
+                Debug.Assert((this.ContainingMemberOrLambda as MethodSymbol)?.MethodKind != MethodKind.LocalFunction);
```

Comment:
BTW, I believe this assert can fail for a local function nested into another local function.